lib/ach/ach_file.rb

```diff
@@ -160,6 +160,15 @@ def parse data
     def parse_descriptive_date(date_string)
       return if date_string.strip.empty?
 
+      date_time =
+        case date_string
+        when 'SD1300' then Date.today + 13.0/24
+        when 'SD1700' then Date.today + 17.0/24
+        when 'SD1800' then Date.today + 18.0/24
+        end
```

Comment:
I'm rusty on ruby, but SDHHMM is hours and minutes. There could be any value from SD0001 to SD2359 in practice.

Reply:
Refactored using a regex (hopefully I don't have two problems now)